crates/p2p/src/swarm/errors.rs

```diff
@@ -8,6 +8,33 @@ use thiserror::Error;
 /// P2P result type.
 pub type P2PResult<T> = Result<T, Error>;
 
+/// Variations of error during a setup handshake.
+#[derive(Debug, Error)]
+pub enum ErrorDuringSetupHandshakeVariations {
```

Comment:
That's a very bad naming.
What about this?

```suggestion
pub enum SetupError {
```

Reply:
And we can rename the `Error` below as `SwarmError` also